src/Microsoft.Framework.Runtime.Hosting/NamedResourceReader.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+
+using System.Collections.Generic;
+using System.IO;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Framework.Runtime
+{
+    internal static class NamedResourceReader
+    {
+        public static IDictionary<string, string> ReadNamedResources(JObject rawProject, string projectFilePath)
+        {
+            var namedResourceToken = rawProject["namedResource"];
+            if (namedResourceToken == null)
+            {
+                return new Dictionary<string, string>();
+            }
+
+            if (namedResourceToken.Type != JTokenType.Object)
+            {
+                throw FileFormatException.Create("Value must be object.", namedResourceToken, projectFilePath);
```

Comment:
Rather than parsing the _json_ object as raw `JObject` how about build a _data model_ class to deserialize the _json_ all together?